File: src/apis/__generated/services/MenuController.ts
import type {Executor} from '../';
import type {MenuDto} from '../model/dto/';
import type {
    MenuInput, 
    MenuSpec, 
    Page, 
    QueryRequest
} from '../model/static/';

export class MenuController {
    
    constructor(private executor: Executor) {}
    
    delete: (options: MenuControllerOptions['delete']) => Promise<
        boolean | undefined
    > = async(options) => {
        let _uri = '/menu/delete';
        return (await this.executor({uri: _uri, method: 'POST', body: options.body})) as Promise<boolean | undefined>;
    }
    
    findById: (options: MenuControllerOptions['findById']) => Promise<
        MenuDto['MenuRepository/COMPLEX_FETCHER']
    > = async(options) => {
        let _uri = '/menu/';
        _uri += encodeURIComponent(options.id);
        return (await this.executor({uri: _uri, method: 'GET'})) as Promise<MenuDto['MenuRepository/COMPLEX_FETCHER']>;
    }
    
    query: (options: MenuControllerOptions['query']) => Promise<
        Page<MenuDto['MenuRepository/COMPLEX_FETCHER']>
    > = async(options) => {
        let _uri = '/menu/query';
        return (await this.executor({uri: _uri, method: 'POST', body: options.body})) as Promise<Page<MenuDto['MenuRepository/COMPLEX_FETCHER']>>;
    }
    
    save: (options: MenuControllerOptions['save']) => Promise<
        string
    > = async(options) => {
        let _uri = '/menu/save';
        return (await this.executor({uri: _uri, method: 'POST', body: options.body})) as Promise<string>;
    }
}

export type MenuControllerOptions = {
    'findById': {
        id: string
    }, 
    'query': {
        body: QueryRequest<MenuSpec>
    }, 
    'save': {
        body: MenuInput
    }, 
    'delete': {
        body: Array<string>
    }
}

File: server/src/main/java/kaigee/top/menu/controller/MenuController.java
package kaigee.top.menu.controller;
import kaigee.top.infrastructure.model.QueryRequest;
import kaigee.top.menu.entity.Menu;
import kaigee.top.menu.entity.dto.MenuInput;
import kaigee.top.menu.entity.dto.MenuSpec;
import kaigee.top.menu.repository.MenuRepository;
import kaigee.top.menu.service.MenuService;
import lombok.AllArgsConstructor;
import org.babyfish.jimmer.client.FetchBy;
import org.babyfish.jimmer.client.meta.DefaultFetcherOwner;
import org.springframework.data.domain.Page;
import org.springframework.validation.annotation.Validated;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequestMapping("menu")
@AllArgsConstructor
@DefaultFetcherOwner(MenuRepository.class)
public class MenuController {
    private final MenuService menuService;

    @GetMapping("{id}")
    public @FetchBy(value = "COMPLEX_FETCHER") Menu findById(@PathVariable String id) {
        return menuService.findById(id);
    }

    @PostMapping("query")
    public Page< @FetchBy(value = "COMPLEX_FETCHER") Menu> query(@RequestBody QueryRequest<MenuSpec> queryRequest) {
        return menuService.query(queryRequest);
    }

    @PostMapping("save")
    public String save(@RequestBody @Validated MenuInput menu) {
        return menuService.save(menu);
    }

    @PostMapping("delete")
    public Boolean delete(@RequestBody List<String> ids) {
        return menuService.delete(ids);
    }



}